frontend/simple/views/components/Chatroom/ChatMain.vue

```diff
@@ -0,0 +1,328 @@
+<template>
+  <div class="c-chatmain" :class="{ isActive: summary.title }">
+    <template v-if="summary.title">
+      <chat-header
+        :title="summary.title"
+        :description="summary.description"
+        routerBack="/messages"
+      >
+        <template slot="actions">
+          <button class="button is-icon">
+            <i class="fa fa-user-plus"></i>
+          </button>
+          <button class="button is-icon">
+            <i class="fa fa-bell"></i>
+          </button>
+          <menu-parent class="level-right">
+            <menu-trigger class="is-icon">
+              <i class="fa fa-ellipsis-v"></i>
+            </menu-trigger>
+
+            <!-- TODO later - be a drawer on mobile -->
+            <menu-content class="c-actions-content">
+              <list hasMargin>
+                <menu-item tag="button" itemId="hash-1" icon="heart">
+                  Option 1
+                </menu-item>
+                <menu-item tag="button" itemId="hash-2" icon="heart">
+                  Option 2
+                </menu-item>
+                <menu-item tag="button" itemId="hash-3" icon="heart">
+                  Option 3
+                </menu-item>
+              </list>
+            </menu-content>
+          </menu-parent>
+        </template>
+      </chat-header>
+
+      <div class="c-body is-flex" :style="bodyStyles">
+        <div v-if="details.isLoading">
+          LOADING... <!-- TODO Design a cool skeleton loading -->
+        </div>
+        <template v-else>
+          <div class="c-body-conversation" ref="conversation">
+            <conversation-greetings />
+            <component
+              v-for="(message, index) in details.conversation"
+              v-bind="getMessageAt[index]"
```

Comment:
A `<component>` uses `v-bind:is="..."`, not `v-bind=`, to my understanding based on the docs.

Reply:
Yes, and `getMessageAt[index]` returns an object where `is` as one of the keys. I've followed the pattern where we [pass properties of an object](https://vuejs.org/v2/guide/components-props.html#Passing-the-Properties-of-an-Object)

Reply:
Ah, I see. Well, then it's OK, however, I still think `getMessageAt[index]` is wrong. It's invalid javascript.

Reply:
invalid javascript, why? It's a `computed property`, it's like accessing an property that in this case is an object.